RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc

```diff
@@ -121,6 +127,48 @@ namespace {
     {
         return std::pair<bool,bool>(df.linkError(), df.capidError());
     }
+
+    std::unique_ptr<HBHEStatusBitSetter> parse_HBHEStatusBitSetter(
+        const edm::ParameterSet psdigi)
+    {
+        return std::make_unique<HBHEStatusBitSetter>(
+            psdigi.getParameter<double>("nominalPedestal"),
+            psdigi.getParameter<double>("hitEnergyMinimum"),
+            psdigi.getParameter<int>("hitMultiplicityThreshold"),
+            psdigi.getParameter<std::vector<edm::ParameterSet> >("pulseShapeParameterSets"));
+    }
+
+    std::unique_ptr<HBHEPulseShapeFlagSetter> parse_HBHEPulseShapeFlagSetter(
+        const edm::ParameterSet &psPulseShape)
+    {
+        return std::make_unique<HBHEPulseShapeFlagSetter>(
+            psPulseShape.getParameter<double>("MinimumChargeThreshold"),
+            psPulseShape.getParameter<double>("TS4TS5ChargeThreshold"),
```

Comment:
To clarify, my original comment about reduction to the number of arguments in function calls was going towards all the code in your purview ( RecoLocalCalo/HcalRec\* etc) if you get to refactoring it.

Specific  to these lines.
http://stackoverflow.com/questions/4592352/number-of-parameters-for-a-constructor
gives the same conclusion that 20-some parameters are too many.
My solution with a struct with as many setting params is not appropriate for the constructor call.


Your ```parse_HBHEPulseShapeFlagSetter(edm::ParameterSet const&) ``` is definitely an improvement though, compared to explicit constructor calls in possibly various places.
this can be made a part of HBHEPulseShapeFlagSetter